File: cv-app/src/services/models/template.ts

import { Control } from "./control";
import { CreatedBy } from "./createdBy";
import { CSSContent } from "./cssContent";
import { Language } from "./language";
import { TemplateType } from "./templateType";

export type Template =
    {
        id: any,
        templateTypeId: any,
        name: any,
        image: any,
        description: any,
        package: any,
        isArchived: boolean,
        totalRows: number,
        templateType: TemplateType,
        version: any,
        author: any,
        controls: Control[],
        cssContents: CSSContent[]
        languages: Language[]
    }

File: CV.API/Domain/Entities/Template.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace CV.API.Domain.Entities
{
    public class Template : BaseEntity
    {
        public string Name { get; set; }
        public string Image { get; set; }
        public string Version { get; set; }
        public string Author { get; set; }
        public string Description { get; set; }
        public string Package { get; set; }
        public Guid TemplateTypeId { get; set; } 
        public bool IsArchived { get; set; } 
        public virtual TemplateType TemplateType { get; set; }
        public virtual List<CSSContent> CSSContents { get; set; }
        public virtual List<Language> Languages { get; set; }
        public virtual List<Control> Controls { get; set; }
        [NotMapped]
        public int TotalRows { get; set; }
    }
}
